lib/converter.js

```diff
@@ -45,51 +63,102 @@ async function convert(inputFile, options, timings) {
     html = doc.convert(options)
   }
   const workingDir = path.dirname(inputFile)
+  const inputFilename = path.basename(inputFile)
   const inputFilenameWithoutExt = path.basename(inputFile, path.extname(inputFile))
   const outputFile = path.join(workingDir, inputFilenameWithoutExt + '.pdf')
+  const tempFilename = inputFilenameWithoutExt + '_temp.html'
+
+  debug('workingDir', workingDir);
+  debug('inputFilename', inputFilename);
+  debug('inputFilenameWithoutExt', inputFilenameWithoutExt);
+  debug('outputFile', outputFile);
+  debug('tempFilename', tempFilename);
+
   let tempFile;
   if (path.isAbsolute(workingDir)) {
-    tempFile = path.join(workingDir, inputFilenameWithoutExt + '_temp.html')
+    tempFile = path.join(workingDir, tempFilename)
   } else {
-    tempFile = path.normalize(path.join(process.cwd(), workingDir, inputFilenameWithoutExt + '_temp.html'))
+    tempFile = path.normalize(path.join(process.cwd(), workingDir, tempFilename))
   }
+
+  debug('tempFile', tempFile);
+
   const puppeteerConfig = {
     headless: true,
-    args: ['--no-sandbox']
+    args: ['--no-sandbox'],
   }
   const browser = await puppeteer.launch(puppeteerConfig);
   const page = await browser.newPage()
+  // page.setDefaultNavigationTimeout(90000)
 
   page.on('pageerror', function (err) {
-    console.log('Page error: ' + err.toString())
-  }).on('error', function (err) {
-    console.log('Error: ' + err.toString())
+    console.error('Page error: ' + err.toString())
+  })
+
+  const [, port] = await Promise.all([
+    writeFile(tempFile, html),
+    getPort(),
+  ])
+
+  const url = `http://localhost:${port}/${tempFilename}`;
+  const httpOptions = {
+    public: workingDir,
```

Comment:
With this, public assets are relative to the original document. It prevents the need for `relativeFilePath` and `baseAbsolutePath `.